[BUG] 4.0.0: build fails by missing linking with libz

```console
[tkloczko@barrel x86_64-redhat-linux-gnu]$ cmake -L
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.


CMake Error: The source directory "/home/tkloczko/rpmbuild/BUILD/warzone2100/x86_64-redhat-linux-gnu" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
-- Cache values
BUILD_EXAMPLES:BOOL=OFF
BUILD_SHARED_LIBS:BOOL=ON
BUILD_TESTING:BOOL=ON
CMAKE_BUILD_TYPE:STRING=RelWithDebInfo
CMAKE_INSTALL_PREFIX:PATH=/home/tkloczko/wz/install
CPACK_COMMAND:FILEPATH=/usr/bin/cpack
CURL_CONFIG_EXECUTABLE:FILEPATH=/usr/bin/curl-config
DISCORDRPC_ENABLE_CLANG_FORMAT:BOOL=OFF
ENABLE_DISCORD:BOOL=ON
ENABLE_DOCS:BOOL=ON
ENABLE_IO_THREAD:BOOL=ON
ENABLE_NLS:BOOL=ON
GLSLC:FILEPATH=/usr/bin/glslc
GZIP_BIN:FILEPATH=/usr/bin/gzip
NO_GETADDRINFO:BOOL=OFF
OpenAL_DIR:PATH=/usr/lib64/cmake/OpenAL
RAPIDJSON:FILEPATH=/usr/include/rapidjson
RE2_BUILD_TESTING:BOOL=OFF
SDL2_DIR:PATH=/usr/lib64/cmake/SDL2
SQLITECPP_BUILD_EXAMPLES:BOOL=OFF
SQLITECPP_BUILD_TESTS:BOOL=OFF
SQLITECPP_INCLUDE_SCRIPT:BOOL=OFF
SQLITECPP_INTERNAL_SQLITE:BOOL=OFF
SQLITECPP_RUN_CPPCHECK:BOOL=OFF
SQLITECPP_RUN_CPPLINT:BOOL=OFF
SQLITECPP_RUN_DOXYGEN:BOOL=OFF
SQLITECPP_USE_ASAN:BOOL=OFF
SQLITECPP_USE_GCOV:BOOL=OFF
SQLITECPP_USE_STACK_PROTECTION:BOOL=ON
SQLITECPP_USE_STATIC_RUNTIME:BOOL=OFF
SQLITE_ENABLE_ASSERT_HANDLER:BOOL=OFF
SQLITE_ENABLE_COLUMN_METADATA:BOOL=OFF
SQLITE_HAS_CODEC:BOOL=OFF
SQLITE_USE_LEGACY_STRUCT:BOOL=OFF
UPNPC_BUILD_SAMPLE:BOOL=OFF
UPNPC_BUILD_SHARED:BOOL=OFF
UPNPC_BUILD_STATIC:BOOL=ON
UPNPC_BUILD_TESTS:BOOL=OFF
UPNPC_NO_INSTALL:BOOL=TRUE
USEPCRE:BOOL=OFF
USE_STATIC_CRT:BOOL=OFF
UTF8PROC_ENABLE_TESTING:BOOL=OFF
UTF8PROC_INSTALL:BOOL=OFF
WARNINGS_AS_ERRORS:BOOL=OFF
WZ_DISTRIBUTOR:STRING=UNKNOWN
WZ_ENABLE_BACKEND_VULKAN:BOOL=ON
WZ_ENABLE_WARNINGS:BOOL=OFF
WZ_ENABLE_WARNINGS_AS_ERRORS:BOOL=OFF
WZ_FINDSDL2_NOCONFIG:BOOL=OFF
sodium_INCLUDE_DIR:PATH=/usr/include
sodium_LIBRARIES:FILEPATH=/usr/lib64/libsodium.so
```
Linkiong fails with:
```console
[tkloczko@barrel x86_64-redhat-linux-gnu]$ make
make: *** No targets specified and no makefile found.  Stop.
[tkloczko@barrel x86_64-redhat-linux-gnu]$ ninja
[1/9] Generating __shouldnotexist.h, autorevision.h, autorevision.cache
-- Skipping copy; output file already exists at: "/home/tkloczko/rpmbuild/BUILD/warzone2100/x86_64-redhat-linux-gnu/build_tools/autorevision.cache"
-- Found Git: /usr/bin/git (found version "2.31.1")
-- Imported revision data from cache file
[2/4] Running utility command for wz2100_doc
-- Finished generating JS docs
[4/4] Linking CXX executable src/warzone2100
FAILED: src/warzone2100
: && /usr/bin/g++ -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -flto=auto -flto-partition=none -fstack-protector-strong -fstack-clash-protection -Wpedantic -Wall -Wextra -fno-common -fno-math-errno -Wcast-align -Wcast-qual -Wctor-dtor-privacy -Winit-self -Woverloaded-virtual -Wstrict-null-sentinel -Wwrite-strings -Wpointer-arith -Wmissing-noreturn -Wnon-virtual-dtor -Wnull-dereference -Wduplicated-cond -Walloc-zero -Walloca-larger-than=1024 -Wrestrict -Wfloat-conversion -Wformat-security -Wstringop-truncation -Wno-logical-op -Wno-float-conversion -Wno-sign-compare -Wno-unused-parameter -Wno-format-truncation -DNDEBUG -Wl,-z,relro -Wl,--as-needed -Wl,--gc-sections -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -flto=auto -flto-partition=none -fuse-linker-plugin     -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now src/CMakeFiles/warzone2100.dir/3rdparty/propertymatcher.cpp.o src/CMakeFiles/warzone2100.dir/action.cpp.o src/CMakeFiles/warzone2100.dir/activity.cpp.o src/CMakeFiles/warzone2100.dir/advvis.cpp.o src/CMakeFiles/warzone2100.dir/ai.cpp.o src/CMakeFiles/warzone2100.dir/animation.cpp.o src/CMakeFiles/warzone2100.dir/astar.cpp.o src/CMakeFiles/warzone2100.dir/atmos.cpp.o src/CMakeFiles/warzone2100.dir/aud.cpp.o src/CMakeFiles/warzone2100.dir/baseobject.cpp.o src/CMakeFiles/warzone2100.dir/bucket3d.cpp.o src/CMakeFiles/warzone2100.dir/challenge.cpp.o src/CMakeFiles/warzone2100.dir/chat.cpp.o src/CMakeFiles/warzone2100.dir/cheat.cpp.o src/CMakeFiles/warzone2100.dir/clparse.cpp.o src/CMakeFiles/warzone2100.dir/cmddroid.cpp.o src/CMakeFiles/warzone2100.dir/combat.cpp.o src/CMakeFiles/warzone2100.dir/component.cpp.o src/CMakeFiles/warzone2100.dir/configuration.cpp.o src/CMakeFiles/warzone2100.dir/console.cpp.o src/CMakeFiles/warzone2100.dir/data.cpp.o src/CMakeFiles/warzone2100.dir/design.cpp.o src/CMakeFiles/warzone2100.dir/difficulty.cpp.o src/CMakeFiles/warzone2100.dir/display.cpp.o src/CMakeFiles/warzone2100.dir/display3d.cpp.o src/CMakeFiles/warzone2100.dir/droid.cpp.o src/CMakeFiles/warzone2100.dir/edit3d.cpp.o src/CMakeFiles/warzone2100.dir/effects.cpp.o src/CMakeFiles/warzone2100.dir/faction.cpp.o src/CMakeFiles/warzone2100.dir/feature.cpp.o src/CMakeFiles/warzone2100.dir/fpath.cpp.o src/CMakeFiles/warzone2100.dir/frontend.cpp.o src/CMakeFiles/warzone2100.dir/game.cpp.o src/CMakeFiles/warzone2100.dir/gateway.cpp.o src/CMakeFiles/warzone2100.dir/geometry.cpp.o src/CMakeFiles/warzone2100.dir/group.cpp.o src/CMakeFiles/warzone2100.dir/hci.cpp.o src/CMakeFiles/warzone2100.dir/hci/build.cpp.o src/CMakeFiles/warzone2100.dir/hci/commander.cpp.o src/CMakeFiles/warzone2100.dir/hci/manufacture.cpp.o src/CMakeFiles/warzone2100.dir/hci/objects_stats.cpp.o src/CMakeFiles/warzone2100.dir/hci/research.cpp.o src/CMakeFiles/warzone2100.dir/ingameop.cpp.o src/CMakeFiles/warzone2100.dir/init.cpp.o src/CMakeFiles/warzone2100.dir/intdisplay.cpp.o src/CMakeFiles/warzone2100.dir/intelmap.cpp.o src/CMakeFiles/warzone2100.dir/intimage.cpp.o src/CMakeFiles/warzone2100.dir/intorder.cpp.o src/CMakeFiles/warzone2100.dir/keybind.cpp.o src/CMakeFiles/warzone2100.dir/keyedit.cpp.o src/CMakeFiles/warzone2100.dir/keymap.cpp.o src/CMakeFiles/warzone2100.dir/level_lexer.cpp.o src/CMakeFiles/warzone2100.dir/levels.cpp.o src/CMakeFiles/warzone2100.dir/lighting.cpp.o src/CMakeFiles/warzone2100.dir/loadsave.cpp.o src/CMakeFiles/warzone2100.dir/loop.cpp.o src/CMakeFiles/warzone2100.dir/main.cpp.o src/CMakeFiles/warzone2100.dir/map.cpp.o src/CMakeFiles/warzone2100.dir/mapdisplay.cpp.o src/CMakeFiles/warzone2100.dir/mapgrid.cpp.o src/CMakeFiles/warzone2100.dir/mechanics.cpp.o src/CMakeFiles/warzone2100.dir/message.cpp.o src/CMakeFiles/warzone2100.dir/miscimd.cpp.o src/CMakeFiles/warzone2100.dir/mission.cpp.o src/CMakeFiles/warzone2100.dir/modding.cpp.o src/CMakeFiles/warzone2100.dir/move.cpp.o src/CMakeFiles/warzone2100.dir/multibot.cpp.o src/CMakeFiles/warzone2100.dir/multigifts.cpp.o src/CMakeFiles/warzone2100.dir/multiint.cpp.o src/CMakeFiles/warzone2100.dir/multijoin.cpp.o src/CMakeFiles/warzone2100.dir/multilimit.cpp.o src/CMakeFiles/warzone2100.dir/multimenu.cpp.o src/CMakeFiles/warzone2100.dir/multiopt.cpp.o src/CMakeFiles/warzone2100.dir/multiplay.cpp.o src/CMakeFiles/warzone2100.dir/multistat.cpp.o src/CMakeFiles/warzone2100.dir/multistruct.cpp.o src/CMakeFiles/warzone2100.dir/multisync.cpp.o src/CMakeFiles/warzone2100.dir/musicmanager.cpp.o src/CMakeFiles/warzone2100.dir/nethelpers.cpp.o src/CMakeFiles/warzone2100.dir/notifications.cpp.o src/CMakeFiles/warzone2100.dir/objects.cpp.o src/CMakeFiles/warzone2100.dir/objmem.cpp.o src/CMakeFiles/warzone2100.dir/oprint.cpp.o src/CMakeFiles/warzone2100.dir/order.cpp.o src/CMakeFiles/warzone2100.dir/pointtree.cpp.o src/CMakeFiles/warzone2100.dir/power.cpp.o src/CMakeFiles/warzone2100.dir/projectile.cpp.o src/CMakeFiles/warzone2100.dir/qtscript.cpp.o src/CMakeFiles/warzone2100.dir/quickjs_backend.cpp.o src/CMakeFiles/warzone2100.dir/radar.cpp.o src/CMakeFiles/warzone2100.dir/random.cpp.o src/CMakeFiles/warzone2100.dir/raycast.cpp.o src/CMakeFiles/warzone2100.dir/research.cpp.o src/CMakeFiles/warzone2100.dir/scores.cpp.o src/CMakeFiles/warzone2100.dir/selection.cpp.o src/CMakeFiles/warzone2100.dir/seqdisp.cpp.o src/CMakeFiles/warzone2100.dir/stats.cpp.o src/CMakeFiles/warzone2100.dir/structure.cpp.o src/CMakeFiles/warzone2100.dir/template.cpp.o src/CMakeFiles/warzone2100.dir/terrain.cpp.o src/CMakeFiles/warzone2100.dir/text.cpp.o src/CMakeFiles/warzone2100.dir/texture.cpp.o src/CMakeFiles/warzone2100.dir/titleui/gamefind.cpp.o src/CMakeFiles/warzone2100.dir/titleui/msgbox.cpp.o src/CMakeFiles/warzone2100.dir/titleui/old.cpp.o src/CMakeFiles/warzone2100.dir/titleui/passbox.cpp.o src/CMakeFiles/warzone2100.dir/titleui/protocol.cpp.o src/CMakeFiles/warzone2100.dir/titleui/titleui.cpp.o src/CMakeFiles/warzone2100.dir/transporter.cpp.o src/CMakeFiles/warzone2100.dir/updatemanager.cpp.o src/CMakeFiles/warzone2100.dir/urlhelpers.cpp.o src/CMakeFiles/warzone2100.dir/urlrequest.cpp.o src/CMakeFiles/warzone2100.dir/version.cpp.o src/CMakeFiles/warzone2100.dir/visibility.cpp.o src/CMakeFiles/warzone2100.dir/warcam.cpp.o src/CMakeFiles/warzone2100.dir/warzoneconfig.cpp.o src/CMakeFiles/warzone2100.dir/wavecast.cpp.o src/CMakeFiles/warzone2100.dir/wrappers.cpp.o src/CMakeFiles/warzone2100.dir/wzapi.cpp.o src/CMakeFiles/warzone2100.dir/wzpropertyproviders.cpp.o src/CMakeFiles/warzone2100.dir/wzscriptdebug.cpp.o src/CMakeFiles/warzone2100.dir/integrations/wzdiscordrpc.cpp.o -o src/warzone2100  -Wl,-rpath,/home/tkloczko/rpmbuild/BUILD/warzone2100/x86_64-redhat-linux-gnu/lib/sdl:/home/tkloczko/rpmbuild/BUILD/warzone2100/x86_64-redhat-linux-gnu/lib/sequence:/home/tkloczko/rpmbuild/BUILD/warzone2100/x86_64-redhat-linux-gnu/lib/sound:/home/tkloczko/rpmbuild/BUILD/warzone2100/x86_64-redhat-linux-gnu/lib/widget:/home/tkloczko/rpmbuild/BUILD/warzone2100/x86_64-redhat-linux-gnu/3rdparty/re2:/home/tkloczko/rpmbuild/BUILD/warzone2100/x86_64-redhat-linux-gnu/3rdparty/SQLiteCpp:/home/tkloczko/rpmbuild/BUILD/warzone2100/x86_64-redhat-linux-gnu/3rdparty/discord-rpc/src:/home/tkloczko/rpmbuild/BUILD/warzone2100/x86_64-redhat-linux-gnu/3rdparty/utf8proc:/home/tkloczko/rpmbuild/BUILD/warzone2100/x86_64-redhat-linux-gnu/lib/ivis_opengl:  lib/exceptionhandler/libexception-handler.a  lib/gamelib/libgamelib.a  lib/netplay/libnetplay.a  lib/sdl/libsdl-backend.so  lib/framework/libframework.a  lib/sequence/libsequence.so  lib/sound/libsound.so  lib/widget/libwidget.so  3rdparty/launchinfo/liblaunchinfo.a  3rdparty/EmbeddedJSONSignature/libEmbeddedJSONSignature.a  3rdparty/libquickjs.a  /usr/lib64/libcurl.so  3rdparty/re2/libre2.so  /usr/lib64/libsqlite3.so  3rdparty/SQLiteCpp/libSQLiteCpp.so.0  3rdparty/discord-rpc/src/libdiscord-rpc.so  /usr/lib64/libphysfs.so  /usr/lib64/libsodium.so  3rdparty/utf8proc/libutf8proc.so.2.4.1  3rdparty/miniupnp/miniupnpc/libminiupnpc.a  lib/ivis_opengl/libivis-opengl.so  3rdparty/libglad.a  /usr/lib64/libsodium.so  /usr/lib64/libsqlite3.so  -pthread  -ldl  -lpthread && :
level_lexer.cpp:232:8: warning: type ‘struct yy_buffer_state’ violates the C++ One Definition Rule [-Wodr]
lex.audp_.c:223: note: a different type is defined in another translation unit
level_lexer.cpp:247:6: note: the first difference of corresponding definitions is field ‘yy_n_chars’
lex.audp_.c:238: note: a field of same name but different type is defined in another translation unit
level_lexer.cpp:232:8: note: type ‘int’ should match type ‘yy_size_t’
audp_parser.tab.cpp:138: warning: type ‘union YYSTYPE’ violates the C++ One Definition Rule [-Wodr]
../lib/framework/strres_parser.h:39: note: a different type is defined in another translation unit
   39 | typedef union YYSTYPE
      |
../lib/gamelib/audp_parser.ypp:41: note: the first difference of corresponding definitions is field ‘fval’
   41 |         float           fval;
      |
strres_parser.ypp:50: note: a field with different name is defined in another translation unit
audp_parser.tab.cpp:348: warning: type ‘union yyalloc’ violates the C++ One Definition Rule [-Wodr]
strres_parser.cpp:328: note: a different type is defined in another translation unit
audp_parser.tab.cpp:351: note: the first difference of corresponding definitions is field ‘yyvs_alloc’
strres_parser.cpp:331: note: a field of same name but different type is defined in another translation unit
audp_parser.tab.cpp:348: note: type ‘union YYSTYPE’ itself violates the C++ One Definition Rule
../3rdparty/quickjs/quickjs.c: In function ‘js_c_function_data_call’:
../3rdparty/quickjs/quickjs.c:5143: warning: potential null pointer dereference [-Wnull-dereference]
 5143 |     if (unlikely(argc < s->length)) {
      |
../3rdparty/quickjs/quickjs.c:5143: warning: potential null pointer dereference [-Wnull-dereference]
../3rdparty/quickjs/quickjs.c:5144: warning: argument to ‘alloca’ may be too large [-Walloca-larger-than=]
 5144 |         arg_buf = alloca(sizeof(arg_buf[0]) * s->length);
      |
../3rdparty/quickjs/libregexp.c: In function ‘lre_exec’:
../3rdparty/quickjs/libregexp.c:2534: warning: argument to ‘alloca’ may be too large [-Walloca-larger-than=]
 2534 |     stack_buf = alloca(alloca_size);
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘allocatePlayers’ at ../src/game.cpp:1820:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 63 bytes from a string of length 63 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘posixExceptionHandler’ at ../lib/exceptionhandler/exceptionhandler.cpp:659:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
strres_parser.cpp: In function ‘strres_parse’:
strres_lexer.cpp:1178: warning: potential null pointer dereference [-Wnull-dereference]
strres_lexer.cpp:1176: warning: potential null pointer dereference [-Wnull-dereference]
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘runContinue’ at ../src/frontend.cpp:214:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 255 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘decideWRF’ at ../src/multiint.cpp:858:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 256 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘NETrecvGAMESTRUCT’ at ../lib/netplay/netplay.cpp:1046:0,
    inlined from ‘readGameStructsList.constprop’ at ../lib/netplay/netplay.cpp:2543:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 63 bytes from a string of length 809 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘NETrecvGAMESTRUCT’ at ../lib/netplay/netplay.cpp:1054:0,
    inlined from ‘readGameStructsList.constprop’ at ../lib/netplay/netplay.cpp:2543:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 39 bytes from a string of length 737 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘NETrecvGAMESTRUCT’ at ../lib/netplay/netplay.cpp:1068:0,
    inlined from ‘readGameStructsList.constprop’ at ../lib/netplay/netplay.cpp:2543:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 39 bytes from a string of length 673 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 39 bytes from a string of length 633 [-Wstringop-truncation]
In function ‘strncpy’,
    inlined from ‘readGameStructsList.constprop’ at ../lib/netplay/netplay.cpp:2552:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 39 bytes from a string of length 39 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘NETsendGAMESTRUCT.constprop’ at ../lib/netplay/netplay.cpp:894:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 63 bytes from a string of length 63 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘NETsendGAMESTRUCT.constprop’ at ../lib/netplay/netplay.cpp:902:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 39 bytes from a string of length 39 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘NETsendGAMESTRUCT.constprop’ at ../lib/netplay/netplay.cpp:921:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 156 bytes from a string of length 156 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘NETsendGAMESTRUCT.constprop’ at ../lib/netplay/netplay.cpp:928:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 39 bytes from a string of length 39 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘NETsendGAMESTRUCT.constprop’ at ../lib/netplay/netplay.cpp:932:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 39 bytes from a string of length 39 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘NETsendGAMESTRUCT.constprop’ at ../lib/netplay/netplay.cpp:936:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 63 bytes from a string of length 63 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘NETsendGAMESTRUCT.constprop’ at ../lib/netplay/netplay.cpp:940:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 254 bytes from a string of length 254 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘resetPlayerConfiguration’ at ../src/multiint.cpp:3395:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 63 bytes from a string of length 127 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
level_lexer.cpp: In function ‘lev_lex’:
level_lexer.cpp:1324: warning: potential null pointer dereference [-Wnull-dereference]
level_lexer.cpp:1322: warning: potential null pointer dereference [-Wnull-dereference]
../3rdparty/quickjs/quickjs.c: In function ‘JS_CallInternal’:
../3rdparty/quickjs/quickjs.c:16338: warning: argument to ‘alloca’ may be too large [-Walloca-larger-than=]
16338 |     local_buf = alloca(alloca_size);
      |
../3rdparty/quickjs/quickjs.c: In function ‘js_call_c_function’:
../3rdparty/quickjs/quickjs.c:16096: warning: argument to ‘alloca’ may be too large [-Walloca-larger-than=]
16096 |         arg_buf = alloca(sizeof(arg_buf[0]) * arg_count);
      |
../3rdparty/quickjs/quickjs.c: In function ‘js_call_bound_function’:
../3rdparty/quickjs/quickjs.c:16207: warning: argument to ‘alloca’ may be too large [-Walloca-larger-than=]
16207 |     arg_buf = alloca(sizeof(JSValue) * arg_count);
      |
../src/quickjs_backend.cpp: In function ‘runMapScript_QuickJS’:
../3rdparty/quickjs/quickjs-limitedcontext.c:35: warning: potential null pointer dereference [-Wnull-dereference]
   35 |                 return NULL;
      |
../3rdparty/quickjs/quickjs-limitedcontext.c:35: warning: potential null pointer dereference [-Wnull-dereference]
../src/qtscript.cpp: In member function ‘loadPlayerScript’:
../3rdparty/quickjs/quickjs.c:2196: warning: potential null pointer dereference [-Wnull-dereference]
 2196 |         return NULL;
      |
../3rdparty/quickjs/quickjs.c:2196: warning: potential null pointer dereference [-Wnull-dereference]
../src/intelmap.cpp: In function ‘intDisplayPIEView’:
../src/mapdisplay.cpp:56: warning: potential null pointer dereference [-Wnull-dereference]
   56 |         if (psResearch->psStat)
      |
../src/mapdisplay.cpp:56: warning: potential null pointer dereference [-Wnull-dereference]
../src/intelmap.cpp:1053: warning: potential null pointer dereference [-Wnull-dereference]
 1053 |                 image = (SWORD)getResearchForMsg(psMessage->pViewData)->iconID;
      |
../src/intelmap.cpp:1053: warning: potential null pointer dereference [-Wnull-dereference]
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘displayRemoteGame’ at ../src/titleui/gamefind.cpp:534:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 63 bytes from a string of length 63 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘displayRemoteGame’ at ../src/titleui/gamefind.cpp:559:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 63 bytes from a string of length 299 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
../src/multiplay.cpp: In function ‘recvSyncRequest’:
../lib/netplay/nettypes.cpp:193: warning: ‘req_id’ may be used uninitialized [-Wmaybe-uninitialized]
  193 |         uint32_t b = (uint32_t)v << 1 ^ (-((uint32_t)v >> 31));
      |
../src/multiplay.cpp:595: note: ‘req_id’ was declared here
  595 |         int32_t req_id, x, y, obj_id, obj_id2, player_id, player_id2;
      |
../lib/netplay/nettypes.cpp:193: warning: ‘x’ may be used uninitialized [-Wmaybe-uninitialized]
  193 |         uint32_t b = (uint32_t)v << 1 ^ (-((uint32_t)v >> 31));
      |
../src/multiplay.cpp:595: note: ‘x’ was declared here
  595 |         int32_t req_id, x, y, obj_id, obj_id2, player_id, player_id2;
      |
../lib/netplay/nettypes.cpp:193: warning: ‘y’ may be used uninitialized [-Wmaybe-uninitialized]
  193 |         uint32_t b = (uint32_t)v << 1 ^ (-((uint32_t)v >> 31));
      |
../src/multiplay.cpp:595: note: ‘y’ was declared here
  595 |         int32_t req_id, x, y, obj_id, obj_id2, player_id, player_id2;
      |
../lib/netplay/nettypes.cpp:193: warning: ‘obj_id’ may be used uninitialized [-Wmaybe-uninitialized]
  193 |         uint32_t b = (uint32_t)v << 1 ^ (-((uint32_t)v >> 31));
      |
../src/multiplay.cpp:595: note: ‘obj_id’ was declared here
  595 |         int32_t req_id, x, y, obj_id, obj_id2, player_id, player_id2;
      |
../lib/netplay/nettypes.cpp:193: warning: ‘player_id’ may be used uninitialized [-Wmaybe-uninitialized]
  193 |         uint32_t b = (uint32_t)v << 1 ^ (-((uint32_t)v >> 31));
      |
../src/multiplay.cpp:595: note: ‘player_id’ was declared here
  595 |         int32_t req_id, x, y, obj_id, obj_id2, player_id, player_id2;
      |
../lib/netplay/nettypes.cpp:193: warning: ‘obj_id2’ may be used uninitialized [-Wmaybe-uninitialized]
  193 |         uint32_t b = (uint32_t)v << 1 ^ (-((uint32_t)v >> 31));
      |
../src/multiplay.cpp:595: note: ‘obj_id2’ was declared here
  595 |         int32_t req_id, x, y, obj_id, obj_id2, player_id, player_id2;
      |
../lib/netplay/nettypes.cpp:193: warning: ‘player_id2’ may be used uninitialized [-Wmaybe-uninitialized]
  193 |         uint32_t b = (uint32_t)v << 1 ^ (-((uint32_t)v >> 31));
      |
../src/multiplay.cpp:595: note: ‘player_id2’ was declared here
  595 |         int32_t req_id, x, y, obj_id, obj_id2, player_id, player_id2;
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘NETfixDuplicatePlayerNames’ at ../lib/netplay/netplay.cpp:772:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 63 bytes from a string of length 63 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘NEThostGame’ at ../lib/netplay/netplay.cpp:3309:0,
    inlined from ‘hostCampaign’ at ../src/multiopt.cpp:329:0,
    inlined from ‘startHost’ at ../src/multiint.cpp:3605:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 63 bytes from a string of length 127 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘NEThostGame’ at ../lib/netplay/netplay.cpp:3324:0,
    inlined from ‘hostCampaign’ at ../src/multiopt.cpp:329:0,
    inlined from ‘startHost’ at ../src/multiint.cpp:3605:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 39 bytes from a string of length 127 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘NEThostGame’ at ../lib/netplay/netplay.cpp:3325:0,
    inlined from ‘hostCampaign’ at ../src/multiopt.cpp:329:0,
    inlined from ‘startHost’ at ../src/multiint.cpp:3605:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 39 bytes from a string of length 127 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘hostCampaign’ at ../src/multiopt.cpp:344:0,
    inlined from ‘startHost’ at ../src/multiint.cpp:3605:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 63 bytes from a string of length 127 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘NETallowJoining’ at ../lib/netplay/netplay.cpp:3117:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 39 bytes from a string of length 39 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
../src/multistruct.cpp: In function ‘recvStructureInfo’:
../lib/netplay/nettypes.cpp:193: warning: ‘droidType’ may be used uninitialized [-Wmaybe-uninitialized]
  193 |         uint32_t b = (uint32_t)v << 1 ^ (-((uint32_t)v >> 31));
      |
../src/multistruct.cpp:316: note: ‘droidType’ was declared here
  316 |         int32_t droidType;
      |
../src/structure.cpp: In function ‘buildStructureDir’:
../src/structure.cpp:1417: warning: potential null pointer dereference [-Wnull-dereference]
 1417 |                                         debug(LOG_ERROR, "Player %u (%s): is building %s at (%d, %d) but found %s already at (%d, %d)",
      |
../src/structure.cpp:1417: warning: potential null pointer dereference [-Wnull-dereference]
../src/structure.cpp:1417: warning: potential null pointer dereference [-Wnull-dereference]
../src/structure.cpp:1417: warning: potential null pointer dereference [-Wnull-dereference]
../src/multistruct.cpp: In function ‘recvBuildFinished.isra’:
../src/multistruct.cpp:128: warning: potential null pointer dereference [-Wnull-dereference]
  128 |                 if (asStructureStats[typeindex].type == psStruct->pStructureType->type)
      |
../src/multistruct.cpp:128: warning: potential null pointer dereference [-Wnull-dereference]
In function ‘strncpy’,
    inlined from ‘UPNP_GetExternalIPAddress’ at ../3rdparty/miniupnp/miniupnpc/upnpcommands.c:322:0,
    inlined from ‘upnp_add_redirect’ at ../lib/netplay/netplay.cpp:1210:0,
    inlined from ‘NETaddRedirects’ at ../lib/netplay/netplay.cpp:1231:0,
    inlined from ‘NETrecvNet’ at ../lib/netplay/netplay.cpp:2008:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 15 bytes from a string of length 127 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘NETjoinGame’ at ../lib/netplay/netplay.cpp:3759:0,
    inlined from ‘joinGameInternalConnect’ at ../src/multiint.cpp:1051:0,
    inlined from ‘joinGameInternal’ at ../src/multiint.cpp:1015:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 63 bytes from a string of length 127 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘__ct ’ at ../src/multiplay.cpp:1123:0,
    inlined from ‘sendToHumanPlayers’ at ../src/chat.cpp:90:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 254 bytes from a string of length 254 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
../src/multigifts.cpp: In function ‘recvAlliance.isra’:
../lib/netplay/nettypes.cpp:193: warning: ‘value’ may be used uninitialized [-Wmaybe-uninitialized]
  193 |         uint32_t b = (uint32_t)v << 1 ^ (-((uint32_t)v >> 31));
      |
../src/multigifts.cpp:561: note: ‘value’ was declared here
  561 |         int32_t value;
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘kf_AddHelpBlip’ at ../src/keybind.cpp:2769:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 254 bytes from a string of length 254 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In member function ‘operator=’,
    inlined from ‘loadLabels.isra’ at ../src/qtscript.cpp:2006:0:
../src/qtscript.h:207: warning: ‘p.subscriber’ may be used uninitialized [-Wmaybe-uninitialized]
  207 | struct LABEL
      |
../src/qtscript.cpp: In member function ‘loadLabels.isra’:
../src/qtscript.cpp:1992: note: ‘p’ declared here
 1992 |                 LABEL p;
      |
In member function ‘operator=’,
    inlined from ‘loadLabels.isra’ at ../src/qtscript.cpp:2038:0:
../src/qtscript.h:207: warning: ‘p.p1’ may be used uninitialized [-Wmaybe-uninitialized]
  207 | struct LABEL
      |
../src/qtscript.cpp: In member function ‘loadLabels.isra’:
../src/qtscript.cpp:1992: note: ‘p’ declared here
 1992 |                 LABEL p;
      |
In member function ‘operator=’,
    inlined from ‘loadLabels.isra’ at ../src/qtscript.cpp:2038:0:
../src/qtscript.h:207: warning: ‘p.p2’ may be used uninitialized [-Wmaybe-uninitialized]
  207 | struct LABEL
      |
../src/qtscript.cpp: In member function ‘loadLabels.isra’:
../src/qtscript.cpp:1992: note: ‘p’ declared here
 1992 |                 LABEL p;
      |
In member function ‘operator=’,
    inlined from ‘loadLabels.isra’ at ../src/qtscript.cpp:2038:0:
../src/qtscript.h:207: warning: ‘p.subscriber’ may be used uninitialized [-Wmaybe-uninitialized]
  207 | struct LABEL
      |
../src/qtscript.cpp: In member function ‘loadLabels.isra’:
../src/qtscript.cpp:1992: note: ‘p’ declared here
 1992 |                 LABEL p;
      |
../src/display3d.cpp: In function ‘draw3DScene’:
../src/display3d.cpp:972: warning: null pointer dereference [-Wnull-dereference]
  972 |                 *crash = 0x3; // deliberate null-dereference
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘writeScriptState’ at ../src/game.cpp:6918:0,
    inlined from ‘saveGame’ at ../src/game.cpp:2868:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘rebuildSearchPath’ at ../src/init.cpp:326:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘rebuildSearchPath’ at ../src/init.cpp:358:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘rebuildSearchPath’ at ../src/init.cpp:361:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘rebuildSearchPath’ at ../src/init.cpp:375:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘rebuildSearchPath’ at ../src/init.cpp:421:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘rebuildSearchPath’ at ../src/init.cpp:424:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘rebuildSearchPath’ at ../src/init.cpp:432:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘rebuildSearchPath’ at ../src/init.cpp:435:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘rebuildSearchPath’ at ../src/init.cpp:294:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘rebuildSearchPath’ at ../src/init.cpp:297:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘rebuildSearchPath’ at ../src/init.cpp:305:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘rebuildSearchPath’ at ../src/init.cpp:308:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘rebuildSearchPath’ at ../src/init.cpp:313:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 4095 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
../src/multiplay.cpp: In function ‘recvMessage.isra’:
../lib/netplay/nettypes.cpp:193: warning: ‘droidType’ is used uninitialized [-Wuninitialized]
  193 |         uint32_t b = (uint32_t)v << 1 ^ (-((uint32_t)v >> 31));
      |
../src/multibot.cpp:382: note: ‘droidType’ was declared here
  382 |                 int32_t droidType;
      |
../lib/netplay/nettypes.cpp:193: warning: ‘sender’ is used uninitialized [-Wuninitialized]
  193 |         uint32_t b = (uint32_t)v << 1 ^ (-((uint32_t)v >> 31));
      |
../src/multiplay.cpp:1623: note: ‘sender’ was declared here
 1623 |         int32_t sender, receiver, locX, locY;
      |
../lib/netplay/nettypes.cpp:193: warning: ‘receiver’ is used uninitialized [-Wuninitialized]
  193 |         uint32_t b = (uint32_t)v << 1 ^ (-((uint32_t)v >> 31));
      |
../src/multiplay.cpp:1623: note: ‘receiver’ was declared here
 1623 |         int32_t sender, receiver, locX, locY;
      |
../lib/netplay/nettypes.cpp:193: warning: ‘locX’ is used uninitialized [-Wuninitialized]
  193 |         uint32_t b = (uint32_t)v << 1 ^ (-((uint32_t)v >> 31));
      |
../src/multiplay.cpp:1623: note: ‘locX’ was declared here
 1623 |         int32_t sender, receiver, locX, locY;
      |
../lib/netplay/nettypes.cpp:193: warning: ‘locY’ is used uninitialized [-Wuninitialized]
  193 |         uint32_t b = (uint32_t)v << 1 ^ (-((uint32_t)v >> 31));
      |
../src/multiplay.cpp:1623: note: ‘locY’ was declared here
 1623 |         int32_t sender, receiver, locX, locY;
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘levLoadData.constprop’ at ../src/levels.cpp:641:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 31 bytes from a string of length 256 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘loadMultiScripts’ at ../src/multiint.cpp:381:0,
    inlined from ‘levLoadData.constprop’ at ../src/levels.cpp:1003:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 255 bytes from a string of length 255 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘intRunWidgets’ at ../src/hci.cpp:951:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 255 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘renderLoop’ at ../src/loop.cpp:234:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 255 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘NETsetGamePassword’ at ../lib/netplay/netplay.cpp:326:0,
    inlined from ‘processMultiopWidgets’ at ../src/multiint.cpp:3778:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 63 bytes from a string of length 63 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘NETsetLobbyOptField’ at ../lib/netplay/netplay.cpp:309:0,
    inlined from ‘processMultiopWidgets’ at ../src/multiint.cpp:3648:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 63 bytes from a string of length 127 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘NETsetLobbyOptField’ at ../lib/netplay/netplay.cpp:315:0,
    inlined from ‘processMultiopWidgets’ at ../src/multiint.cpp:3835:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 39 bytes from a string of length 63 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘run’ at ../src/multiint.cpp:4469:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 127 bytes from a string of length 127 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘run’ at ../src/multiint.cpp:4483:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 127 bytes from a string of length 127 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘NETsetLobbyOptField’ at ../lib/netplay/netplay.cpp:315:0,
    inlined from ‘run’ at ../src/multiint.cpp:4459:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 39 bytes from a string of length 63 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
In function ‘strncpy’,
    inlined from ‘strlcpy’ at .././lib/framework/string_ext.h:128:0,
    inlined from ‘sstrcpy’ at .././lib/framework/string_ext.h:191:0,
    inlined from ‘loadOK’ at ../src/frontend.cpp:416:0,
    inlined from ‘runSinglePlayerMenu’ at ../src/frontend.cpp:480:0:
/usr/include/bits/string_fortified.h:95: warning: ‘__builtin_strncpy’ output may be truncated copying 255 bytes from a string of length 4095 [-Wstringop-truncation]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |
/usr/bin/ld: /tmp/cc40Tgpt.lto.o: undefined reference to symbol 'deflate'
/usr/bin/ld: /usr/lib64/libz.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```
I made test and took last gcc line and added at the end `-lz` and was able to link main executable maually.
I don't see anywhere in any CMakeLists.txt files detection of the zlib add adding `-lz` to linker options.

BTW all above linker warnings are exposed by using LTO.
It woild be good to clean those warning as well :)

Standard zlib does not provides cmake `FindZlib.cmake` module.
So in my case cmake fails with that patch with:
```console
-- Performing Test COMPILER_SUPPORTS_WNO_PEDANTIC - Success
CMake Error at lib/netplay/CMakeLists.txt:33 (find_package):
  By not providing "FindZlib.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Zlib", but
  CMake did not find one.

  Could not find a package configuration file provided by "Zlib" with any of
  the following names:

    ZlibConfig.cmake
    zlib-config.cmake

  Add the installation prefix of "Zlib" to CMAKE_PREFIX_PATH or set
  "Zlib_DIR" to a directory containing one of the above files.  If "Zlib"
  provides a separate development package or SDK, be sure it has been
  installed.
```
I think that better would be use pkgconfig.
Moment will try to prepare PR for that.

--- a/lib/netplay/CMakeLists.txt
+++ b/lib/netplay/CMakeLists.txt
@@ -30,6 +30,7 @@ file(GLOB HEADERS "*.h")
 file(GLOB SRC "*.cpp")
 
 find_package (Threads REQUIRED)
+find_package (ZLIB REQUIRED)
 
 add_library(netplay STATIC ${HEADERS} ${SRC} "${_netplay_config_output_file}")
 add_dependencies(netplay autorevision_netcodeversion)
@@ -48,7 +49,7 @@ endif()
 if("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
 	target_link_libraries(netplay PRIVATE pthread)
 endif()
-target_link_libraries(netplay PRIVATE Threads::Threads)
+target_link_libraries(netplay PRIVATE Threads::Threads ZLIB::ZLIB)
 if(MSVC)
 	# C4267: 'conversion': conversion from 'type1' to 'type2', possible loss of data // FIXME!!
 	target_compile_options(netplay PRIVATE "/wd4267")

